src/Symfony/Component/Messenger/Tests/Exception/HandlerFailedExceptionTest.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace Symfony\Component\Messenger\Tests\Exception;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Messenger\Envelope;
+use Symfony\Component\Messenger\Exception\HandlerFailedException;
+
+class HandlerFailedExceptionTest extends TestCase
+{
+    public function testThatStringErrorCodeConvertsToInteger()
+    {
+        $envelope = new Envelope(new \stdClass());
+
+        $pdoException = new class() extends \RuntimeException {
+            public function __construct()
+            {
+                $this->code = 'HY000';
+                $this->message = 'test';
+                // no to call parent constructor, it will fail with string error code
+            }
+        };
+
+        $exceptions = [$pdoException];
+
+        $handlerException = new HandlerFailedException($envelope, $exceptions);
+
+        $this->assertIsInt(
+            $handlerException->getCode(),
+            'All exception codes must converts to int'
+        );
```

Comment:
Small nitpick: can be on one line (same for the next one)

Reply:
done

Reply:
forgot to push?

Reply:
[Done for all asserts](https://github.com/symfony/symfony/pull/33713/files#diff-7a30e63fcf1a036cde8c651fb3fd664cR27)